highfive/tests/test_newpr.py

```diff
@@ -18,19 +18,35 @@ def test_find_reviewer(self):
         found_cases = (
             ('r? @foo', 'foo'),
             ('R? @foo', 'foo'),
-            ('....@!##$@#%r? @foo', 'foo'),
-            ('r?:-:-:- @foo', 'foo'),
-            ('Lorem ipsum dolor sit amet, r?@foo consectetur', 'foo'),
+            ('r?: @foo', 'foo'),
+            ('r?- @foo', 'foo'),
+            ('Lorem ipsum dolor sit amet, r? @foo consectetur', 'foo'),
             ('r? @8iAke', '8iAke'),
-            ('r? @D--a--s-h', 'D--a--s-h'),
+            ('r? @D-a-s-h', 'D-a-s-h'),
             ('r? @foo$', 'foo'),
+            (
+                'r? @012345678901234567890123456789012345678',
+                '012345678901234567890123456789012345678'
+            ),
+            (
+                'r? @0-2-4-6-8-0-2-4-6-8-0-2-4-6-8-0-2-4-6-8',
+                '0-2-4-6-8-0-2-4-6-8-0-2-4-6-8-0-2-4-6-8'
+            ),
         )
         not_found_cases = (
             'rr? @foo',
             'r @foo',
             'r?! @foo',
+            'r?@foo',
```

Comment:
@nrc: How do you feel about this? The current regular expression will match the case where there is no whitespace between '?' and '@', but I think it's better to require that whitespace. If you are aware of users that do not include the whitespace, though, that's a strong case to not require it.

Reply:
I'd prefer to be more flexible - there is very little downside to over-matching, and for new contributors, I'd like to Just Work as much as possible.

Reply:
Considering that sentiment, I am going to scrap most of the PR. I'm not really adding anything here except less flexibility. :grinning:

As it's written now, I believe that providing an incorrect username will lead to the appearance of highfive ignoring the user (when, in fact, it just quietly gives up), but that's something we could fix later with a comment from highfive or just falling back to choosing someone using the configs. I think that is what's meant by "close this loophole" in #39.

Reply:
Yes, that's what I meant by 'close this loophole'